bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/form/tempusdominus/AbstractTempusDominusWithIcon.java

```diff
@@ -0,0 +1,160 @@
+package de.agilecoders.wicket.extensions.markup.html.bootstrap.form.tempusdominus;
+
+import de.agilecoders.wicket.extensions.markup.html.bootstrap.icon.FontAwesomeSettings;
+
+import java.util.List;
+
+import org.apache.wicket.Application;
+import org.apache.wicket.Component;
+import org.apache.wicket.MarkupContainer;
+import org.apache.wicket.behavior.AttributeAppender;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.markup.html.form.FormComponent;
+import org.apache.wicket.markup.html.form.FormComponentPanel;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.util.convert.ConversionException;
+import org.apache.wicket.util.convert.IConverter;
+import org.apache.wicket.util.string.StringValue;
+
+import de.agilecoders.wicket.core.markup.html.bootstrap.image.Icon;
+import de.agilecoders.wicket.core.markup.html.bootstrap.image.IconType;
+
+/**
+ * Tempus-dominus with calendar icon.
+ *
+ */
+public abstract class AbstractTempusDominusWithIcon<T> extends FormComponentPanel<T> {
+    private static final long serialVersionUID = 1L;
+
+    private TempusDominusConfig config;
+    private FormComponent<T> dateInput;
+
+    /**
+     * Construct.
+     *
+     * @param markupId markup id
+     * @param config   DateTimePicker config
+     */
+    public AbstractTempusDominusWithIcon(String markupId, TempusDominusConfig config) {
+        this(markupId, null, config);
+    }
+
+    /**
+     * Construct.
+     *
+     * @param markupId markup id
+     * @param model    model
+     * @param config   DateTimePicker config
+     */
+    public AbstractTempusDominusWithIcon(String markupId, IModel<T> model, TempusDominusConfig config) {
+        super(markupId, model);
+        this.config = config;
+    }
+
+    private FormComponent<T> getDateInput() {
+        if (dateInput == null) {
+            dateInput = newInput("date", config.getFormat());
+            dateInput.setModel(getModel());
+        }
+        return dateInput;
+    }
+
+    @Override
+    public void convertInput() {
+        final IConverter<T> converter = getDateInput().getConverter(getType());
+
+        try
+        {
+            T obj = converter.convertToObject(getRawInput(), getLocale());
+            getDateInput().setConvertedInput(obj);
+            setConvertedInput(obj);
+        }
+        catch (ConversionException e)
+        {
+            error(newValidationError(e));
+        }
+    }
+
+    @Override
+    protected List<StringValue> getParameterValues(String inputName) {
+        return super.getParameterValues(inputName + ":date");
```

Comment:
@martin-g is it OK to make hacks like this? :)

Reply:
I think it is OK!